### PR TITLE
authz: add e2e tests for egress gateway

### PR DESCRIPTION
**Pull request by yangminzhu**
https://github.com/istio/istio/issues/19894

This PR adds e2e tests for authorization policy on egress gateway.

Fixes https://github.com/istio/istio/issues/21001

### CODE REVIEW

File: tests/integration/security/testdata/rbac/v1beta1-egress-gateway.yaml.tmpl [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        request:
+          add:
+            x-egress-test: "handled-by-egress-gateway"
+
+---
+
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: service-entry-company
```
Comment (howardjohn) [inline tests/integration/security/testdata/rbac/v1beta1-egress-gateway.yaml.tmpl]: is this used?

Comment (yangminzhu) [reply tests/integration/security/testdata/rbac/v1beta1-egress-gateway.yaml.tmpl]: I think it's not used, removed from the PR.

**Comment by yangminzhu** [comment]
Hmm, it's a bit strange, the test passed when running locally but one of the cases failed with 404 when running in prow, need to investigate.

**Pull request merged**